llvm/utils/TableGen/DXILEmitter.cpp

```diff
@@ -124,32 +106,29 @@ DXILOperationDesc::DXILOperationDesc(const Record *R) {
   // the comment before the definition of class LLVMMatchType in
   // llvm/IR/Intrinsics.td
   SmallVector<int> OverloadParamIndices;
-  for (unsigned i = 0; i < TypeRecsSize; i++) {
-    auto TR = TypeRecs[i];
+  for (unsigned I = 0; I < ParamTypeRecsSize; I++) {
+    auto TR = ParamTypeRecs[I];
     // Track operation parameter indices of any overload types
-    auto isAny = TR->getValueAsInt("isAny");
-    if (isAny == 1) {
+    auto IsAny = TR->getValueAsInt("isAny");
+    if (IsAny == 1) {
       // TODO: At present it is expected that all overload types in a DXIL Op
       // are of the same type. Hence, OverloadParamIndices will have only one
       // element. This implies we do not need a vector. However, until more
       // (all?) DXIL Ops are added in DXIL.td, a vector is being used to flag
       // cases this assumption would not hold.
       if (!OverloadParamIndices.empty()) {
-        bool knownType = true;
+        bool KnownType = true;
         // Ensure that the same overload type registered earlier is being used
         for (auto Idx : OverloadParamIndices) {
-          if (TR != TypeRecs[Idx]) {
-            knownType = false;
+          if (TR != ParamTypeRecs[Idx]) {
+            KnownType = false;
             break;
           }
         }
-        if (!knownType) {
```

Comment:
I see you added an assert( ) here.  Did this changed the behavior?  Before it would report an error using report_fatal_error( ) and have text.  Is the assert( ) doing the same thing?

Reply:
There seem to be a few places where error reporting or llvm_unreachable's have been turned into asserts.  Is there an overall strategy here?

Reply:
> I see you added an assert( ) here. Did this changed the behavior? Before it would report an error using report_fatal_error( ) and have text. Is the assert( ) doing the same thing?

Yes, the error text would be printed.

Reply:
> There seem to be a few places where error reporting or llvm_unreachable's have been turned into asserts. Is there an overall strategy here?

The overall strategy suggested in earlier feedback on this PR was to use `assert` to flag violations of assumptions made in code - for example, in this case, the assumption specified in the TODO comment a few lines above.